primitives/darwinia-core/src/lib.rs

```diff
@@ -37,23 +37,43 @@ mod copy_paste_from_darwinia {
 		OpaqueExtrinsic, Perbill,
 	};
 
+	/// An index to a block.
 	pub type BlockNumber = u32;
+	/// Hashing type
 	pub type Hashing = BlakeTwo256;
+	/// A hash of some data used by the chain.
 	pub type Hash = H256;
+	/// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
 	pub type Signature = EthereumSignature;
+	/// Account public key
 	pub type AccountPublic = <Signature as Verify>::Signer;
+	/// Some way of identifying an account on the chain.
 	pub type AccountId = <AccountPublic as IdentifyAccount>::AccountId;
+	/// Alias to the AccountId type.
 	pub type Address = AccountId;
+	/// Nonce of an account in the chain.
 	pub type Nonce = u32;
+	/// Balance type.
 	pub type Balance = u128;
+	/// Abstraction over the  darwinia block header.
 	pub type Header = generic::Header<BlockNumber, Hashing>;
+	/// Abstraction over the darwinia block.
 	pub type OpaqueBlock = generic::Block<Header, OpaqueExtrinsic>;
 
-	pub const AVERAGE_ON_INITIALIZE_RATIO: Perbill = Perbill::from_perthousand(25);
+	/// We assume that ~5% of the block weight is consumed by `on_initialize` handlers. This is
+	/// used to limit the maximal weight of a single extrinsic.
+	pub const AVERAGE_ON_INITIALIZE_RATIO: Perbill = Perbill::from_percent(5);
+	/// We allow `Normal` extrinsics to fill up the block up to 75%, the rest can be used by
+	/// `Operational` extrinsics.
 	pub const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
-	// TODO: https://github.com/paritytech/parity-bridges-common/issues/1543 - remove `set_proof_size`
-	pub const MAXIMUM_BLOCK_WEIGHT: Weight =
-		Weight::from_ref_time(WEIGHT_REF_TIME_PER_SECOND).set_proof_size(1_000).saturating_mul(2);
+	/// We allow for 0.5 of a second of compute with a 12 second average block time.
+	pub const WEIGHT_MILLISECS_PER_BLOCK: u64 = 500;
+	/// Maximum PoV size we support right now.(Copied from the polkadot repo)
```

Comment:
Then, you should not put this inside the `mod copy_paste_from_darwinia `.

Reply:
Makes a bit of sense. expanded.